sdk/python/jobs/pipelines/1k_demand_forecasting_with_pipeline_components/automl-forecasting-demand-many-models-in-pipeline/automl-forecasting-demand-many-models-in-pipeline.ipynb

```diff
@@ -0,0 +1,1477 @@
+    "\n",
+    "try:\n",
+    "    # Retrieve an already attached Azure Machine Learning Compute.\n",
+    "    compute_target = ml_client.compute.get(amlcompute_cluster_name)\n",
+    "except ResourceNotFoundError as e:\n",
+    "    compute_target = AmlCompute(\n",
+    "        name=amlcompute_cluster_name,\n",
+    "        size=\"STANDARD_DS15_V2\",\n",
+    "        type=\"amlcompute\",\n",
+    "        min_instances=0,\n",
+    "        max_instances=5,\n",
+    "        idle_time_before_scale_down=600,\n",
+    "    )\n",
+    "    poller = ml_client.begin_create_or_update(compute_target)\n",
+    "    poller.wait()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 4. Data\n",
+    "\n",
+    "For illustration purposes we use the UCI electricity data ([link](https://archive.ics.uci.edu/ml/datasets/ElectricityLoadDiagrams20112014#)). The original dataset contains electricity consumption data for 370 consumers measured at 15 minute intervals. In the data set for this demonstrations, we have aggregated to an hourly frequency and converted to the kilowatt hours (kWh) for 10 customers.\n",
+    "\n",
+    "The data for this notebook is located in the `automl-sample-notebook-data` container in the datastore and is publicly available. In the next few cells, we will download the train, test and inference datasets from the public datastore and store them locally in the _parquet_ format."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data_path = \"https://automlsamplenotebookdata.blob.core.windows.net/automl-sample-notebook-data/uci_electro_small_public_mm_train/uci_electro_small_mm_train.csv\"\n",
+    "test_data_path = \"https://automlsamplenotebookdata.blob.core.windows.net/automl-sample-notebook-data/uci_electro_small_public_mm_test/uci_electro_small_mm_test.csv\"\n",
+    "inference_data_path = \"https://automlsamplenotebookdata.blob.core.windows.net/automl-sample-notebook-data/uci_electro_small_public_mm_infer/uci_electro_small_mm_inference.csv\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "time_column_name = \"datetime\"\n",
+    "target_column_name = \"usage\"\n",
+    "time_series_id_column_names = [\"customer_id\"]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_folder_and_save_as_parquet(file_uri, output_folder):\n",
+    "    os.makedirs(output_folder, exist_ok=True)\n",
+    "    data_frame = pd.read_csv(file_uri, parse_dates=[time_column_name])\n",
+    "    file_name = file_uri.split(\"/\")[-1].replace(\".csv\", \".parquet\")\n",
+    "    data_path = os.path.join(output_folder, file_name)\n",
+    "    data_frame.to_parquet(data_path, index=False)\n",
+    "    return None\n",
```

Comment:
As far as you still will read data frame in the next cell, let us return the data frame.
```python
return data_frame
```